File: guess-the-number/guess-the-number.hs
import System.Random
import Text.Printf

from, to :: Int
from = 1
to = 100

main :: IO ()
main = do target <- randomRIO (from, to)
          printf "Guess a number between %d and %d\n" from to
          askNumber target
    where askNumber target = do guess <- readLn
                                case compare guess target of
                                  EQ -> putStrLn "Correct! You got it!"
                                  LT -> do putStrLn "higher"
                                           askNumber target
                                  GT -> do putStrLn "lower"
                                           askNumber target

File: guess-the-number/guess-the-number.rb
from = 1
to = 100

target = rand from..to
puts "Guess a number between #{from} and #{to}"

loop do
  guess = gets.to_i

  if guess == target
    puts "Correct! You got it!"
    break
  elsif guess < target
    puts "higher"
  else
    puts "lower"
  end
end
